### ISSUE TITLE
Support for PiZero/PiZeroW

### ISSUE THREAD

**Issue by bdurrer** [jens-maus/RaspberryMatic#69]
The RaspberryMatic and ELV state that the module/software is compatible with Pi 2 & 3.
However, the Zero and the Zero Wireless both have the same GPIO pinout, so I don't see why they should not work?

Is the missing support maybe an oversight or is there an obivous reason why I shouldn't try it?

**Comment by jens-maus** [comment]
This is of course a very valid question (which in fact I have already thought about myself). The answer is, however, also quite straight forward:

1. Technically there shouldn't be any reasons why the Pi Zero or the newer Pi Zero W won't work with RaspberryMatic – but there is still some work to do!
2. The currently shipped image won't work with the Pi Zero hardware out-of-the-box because the main hardware on the Pi Zero is based on the older RaspberryPi1 model which has a different/older SoC chip than the Pi2 or Pi3 models.
3. It shouldn't be any big deal to actually tune the build environment of RaspberryMatic to also build an own image for the Pi Zero / Pi Zero W which should then also enable RaspberryMatic to work on the much older RaspberryPi1 models.
4. I simply lacked the time (though I have a Pi Zero at home) to actually work on this.
5. The sold numbers of Pi Zero boards are still quite small (only one board per customer limitation)
6. The older hardware (compared to a Pi3) and the point that a CCU should normally run as a server 24h/7days also raises some questions if there are really big numbers to be expected for a CCU running on a PiZero. 
7. Up to the recent release of the PiZeroW with embedded WiFi/Bluetooth it even was more questionable to use a PiZero at all for RaspberryMatic because until recently the WiFI support was non existent in RaspberryMatic.
8. Now that we have working WiFi-support in RaspberryMatic (coming with the next beta5 version) the PiZero might be more usable. But also keep in mind that using WiFi for something critical like a CCU is also questionable in itself due to stability/security concerns.

To summarize: I fully understand the wish to use a PiZero for RaspberryMatic and I haven't yet abandoned the idea completly. But it simply had and still has low priority to be implemented. But if you have something to contribute, feel free to send over some Pull Requests which I would happily integrate in the main repository of RaspberryMatic :)

Please also note that the same question can be raised for other embedded platforms like BananaPi, etc. – which I would in principle also be happy to support in a future version of RaspberryMatic.

**Comment by bdurrer** [comment]
Thanks for the response.

You can order more than one ZeroW by now and it's way more useful thant the old Zero, so I guess the sales will go up.
Funny, until you mentioned the WLAN I didn't even think about the fact that it does not have a LAN port 😄 But I guess that's not an issue [in my case] with a stable WLAN.

I'm a newbie in the Homematic world for now, but since I'll get some ZeroWs soon I sure can try to build it on one of them.

**Comment by renne** [comment]
The HM-MOD-RPI-PCB works on the GPIO of Model B (without +) with YAHM. What's the state of the RaspberryPi Model B in Raspberrymatic?

**Comment by jens-maus** [comment]
@renne Not working yet. I guess it will be supported once I have support for the Pi Zero line ready. However, support for other RaspberryPi models than Pi2/3 has currently low priority.

**Comment by ChrisLechn** [comment]
Any way we can support with the developement for the PI Zero? I would love to see a working image for my PI ZeroW to be able to use Homematic in my caravan. :-)

**Comment by renne** [comment]
@jens-maus RaspberryPi Zero W + HM-MOD-RPI-PCB would be a cheap replacement for the LAN Gateway (HM-LGW-O-TW-W-EU). ;)

**Comment by nanosonde** [comment]
Would it be possible to generate an additional image for the RasperryPi Zero W that only includes the hmlangw daemon? So one could easily build a wireless LAN GW that does not need any configuration.

**Comment by jens-maus** [comment]
Of course this is possible and it is also a plan. However, I need time for that which I currently don't have. But feel free to help out by implementing Pi Zero W support :)

**Comment by nanosonde** [comment]
Is there anything special concerning Pi Zero support in the "pizero" branch yet?

**Comment by jens-maus** [comment]
No, there is nothing "special" about it, it is just a development branch which I haven't finished yet because other things were more important. Thus, the "pizero" branch doesn't produce a running RaspberryMatic for PiZero systems (yet).
